lib/path_expander.rb

```diff
@@ -154,9 +154,10 @@ def filter_files files, ignore
 
       only_paths = File::FNM_PATHNAME
       files = files.reject { |f|
-        dirs.any?     { |i| File.fnmatch?(i, File.dirname(f), only_paths) } ||
-          globs.any?  { |i| File.fnmatch?(i, f) } ||
-          ifiles.any? { |i| File.fnmatch?(i, f, only_paths) }
+        f = File.expand_path(f)
+        dirs.any?     { |i| File.fnmatch?(File.expand_path(i), File.dirname(f), only_paths) } ||
+          globs.any?  { |i| File.fnmatch?(File.expand_path(i), f) } ||
+          ifiles.any? { |i| File.fnmatch?(File.expand_path(i), f, only_paths) }
```

Comment:
That `any?` cluster is already complex enough. I think I'm going to `map!` the three collections beforehand.